draft-ietf-quic-transport.md

```diff
@@ -2241,11 +2242,21 @@ linked by any other entity.
 At any time, endpoints MAY change the Destination Connection ID they send to a
 value that has not been used on another path.
 
-An endpoint MUST use a new connection ID if it initiates connection migration as
-described in {{initiating-migration}} or probes a new network path as described
-in {{probing}}.  An endpoint MUST use a new connection ID in response to a
-change in the address of a peer if the packet with the new peer address uses an
-active connection ID that has not been previously used by the peer.
+An endpoint MUST NOT reuse a connection ID when sending from more than one local
+address, for example when initiating connection migration as described in
+{{initiating-migration}} or when probing a new network path as described in
+{{probing}}.
+
+Similarly, an endpoint MUST NOT reuse a connection ID when sending to more than
+one destination address, for example when responding to a change in the address
+of a peer if the packet with the new peer address uses an active connection ID
+that has not been previously used by the peer.
```

Comment:
This is a little long and the 'for example' is a little buried.

```suggestion
Similarly, an endpoint MUST NOT reuse a connection ID when sending to a remote
address from which the peer has chosen to use a connection ID that has not
previously been used.  This occurs when an endpoint receives packets that
include a change to both the source address and destination connection ID at the
same time.
```

Reply:
I'm happy with Mike's suggestion.

Reply:
Is rationale useful here?   ie "as this indicates the peer did not intentionally change its address."?

Reply:
Thanks! This is nice wording, added in 5509f1d.
 
Re: rationale, I think that's mostly covered by two notes indicating that this can happen.
This one, the immediate next paragraph after the sentence in question:
```
These requirements regarding connection ID reuse apply only to the sending of
packets, as unintentional changes in path without a change in connection ID are
possible.  For example, after a period of network inactivity, NAT rebinding
might cause packets to be sent on a new path when the client resumes sending.
An endpoint responds to such an event as described in {{migration-response}}.
```
and, immediately after the other place we talk about this kind of interaction: 
```
Not all changes of peer address are intentional, or active, migrations. The peer
could experience NAT rebinding: a change of address due to a middlebox, usually
a NAT, allocating a new outgoing port or even a new outgoing IP address for a
flow.  An endpoint MUST perform path validation ({{migrate-validate}}) if it
detects any change to a peer's address, unless it has previously validated that
address.
```